### PR TITLE
[stable/external-dns] Create secret for google service account

### PR DESCRIPTION
**Pull request by tetianakravchenko**
#### What this PR does / why we need it: 

To manage secrets `credentials.json` holding google service account json file within external-dns helm chart.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by linki** [comment]
@tetianakravchenko Thanks! Can you please also sign the remaining commits to make the DCO check happy.

> 2 out of 3 commits are missing signoff

**Comment by tetianakravchenko** [comment]
@linki DCO check is happy now, also conflicts were resolved

**Comment by linki** [comment]
Thanks @tetianakravchenko

**Comment by linki** [comment]
> Chart doesn't have maintainers

We should fix that

**Comment by tetianakravchenko** [comment]
@linki error:
> Chart doesn't have maintainers

should be fixed now, could you please have another look?

**Comment by linki** [comment]
@tetianakravchenko Could you fix the merge conflict?

**Comment by tetianakravchenko** [comment]
@linki merge conflict was fixed, could you please have another look?

File: stable/external-dns/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ description:
   Configure external DNS servers (AWS Route53, Google CloudDNS and others)
   for Kubernetes Ingresses and Services
 name: external-dns
-version: 1.3.1
+version: 1.3.3
```
Comment (linki) [inline stable/external-dns/Chart.yaml]: It looks like the current version is `1.3.1`(https://github.com/helm/charts/blob/master/stable/external-dns/Chart.yaml#L6) so this should be `1.3.2`.

Comment (tetianakravchenko) [reply stable/external-dns/Chart.yaml]: fixed https://github.com/helm/charts/pull/8882/commits/2b47e971f9f84007d3abd033ad8036cb08457435

**Comment by linki** [comment]
@tetianakravchenko Thank  you!

/lgtm

**Comment by linki** [comment]
@tetianakravchenko Could you double-check that this also works fine when `serviceAccountKey` is not defined in `values.yaml`. If not it would break for exiting users.

**Comment by sebastien-prudhomme** [comment]
Got an error with this PR:

```
template: external-dns/charts/external-dns/templates/secret.yaml:14:12: executing "external-dns/charts/external-dns/templates/secret.yaml" at <eq .Values.provider ...>: error calling eq: invalid type for comparison
```

Perhaps it should use:

```diff
@@ -11,7 +11,7 @@ data:
   credentials: {{ include "external-dns.aws-credentials" . | b64enc | quote }}
   config: {{ include "external-dns.aws-config" . | b64enc | quote }}
 {{- end}}
-{{- if and (eq .Values.provider "google" .Values.google.serviceAccountKey) }}
+{{- if and (eq .Values.provider "google") .Values.google.serviceAccountKey }}
   credentials.json: {{ .Values.google.serviceAccountKey | b64enc | quote }}
 {{- end}}
 {{- if .Values.cloudflare.apiKey }}
```

**Comment by tetianakravchenko** [comment]
@sebastien-prudhomme I've tried to reproduce the issue, but couldn't (with helm version 2.9.1 and 2.11.0 and with set `.Values.google.serviceAccountKey` and `.Values.provider`), might it be related to this issue https://github.com/helm/helm/issues/2979 ?

**Comment by tetianakravchenko** [comment]
@linki `serviceAccountKey` is not defined as `required`, shouldn't break existing setup. Also before every usage of `.Values.google.serviceAccountKey` the existence of this values is checked in if condition (e.g. https://github.com/tetianakravchenko/charts/blob/master/stable/external-dns/templates/deployment.yaml#L93-L96, https://github.com/tetianakravchenko/charts/blob/master/stable/external-dns/templates/deployment.yaml#L178-L186), but for consistency reason, I guess it would make sense to set it to default `values.yaml` as "", should I make another PR?
I've tried to install with values:
```
provider: google

google:
  project: test
  serviceAccountSecret: test-secrets
```
was rendered and installed without issues

**Comment by tetianakravchenko** [comment]
@sebastien-prudhomme I've checked for `aws` provider, I got the same error, here is the PR https://github.com/helm/charts/pull/10487

**Comment by sebastien-prudhomme** [comment]
@tetianakravchenko your new PR is working fine with my provider (openstack), thanks

**Comment by linki** [comment]
@tetianakravchenko Thank you

**Pull request merged**